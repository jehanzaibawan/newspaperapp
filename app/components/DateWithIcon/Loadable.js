/**
 *
 * Asynchronously loads the component for DateWithIcon
 *
 */

import loadable from 'utils/loadable';

export default loadable(() => import('./index'));
