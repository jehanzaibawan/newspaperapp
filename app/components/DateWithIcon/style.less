.npApp_date-with-icon {
  display: flex;
  font-size: 0.8rem;
  color: #a6a6a6;

  span:before {
    font-family: 'all-icon-font';
    display: inline-flex;
    content: '\E914';
    color: #767676;
    padding: 0 0.5rem;
    font-weight: bolder;
  }
}