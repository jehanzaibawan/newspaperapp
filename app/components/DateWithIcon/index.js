/**
 *
 * DateWithIcon
 *
 */

import React, { memo } from 'react';
import PropTypes from 'prop-types';
import './style.less';

const DateWithIcon = props => (
  <div className="npApp_date-with-icon">
    <span />
    {props.publishedOn}
  </div>
);

DateWithIcon.propTypes = {
  publishedOn: PropTypes.string.isRequired,
};

export default memo(DateWithIcon);
