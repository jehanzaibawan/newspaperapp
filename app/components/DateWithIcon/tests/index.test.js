/**
 *
 * Tests for DateWithIcon
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import DateWithIcon from '../index';

describe('<DateWithIcon />', () => {
  it('Should render and match the snapshot', () => {
    const props = {
      publishedOn: '2016-01-01',
    };

    const {
      container: { firstChild },
    } = render(<DateWithIcon {...props} />);
    expect(firstChild).toMatchSnapshot();
  });
});
