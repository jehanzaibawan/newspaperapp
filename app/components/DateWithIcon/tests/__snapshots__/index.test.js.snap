// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DateWithIcon /> Should render and match the snapshot 1`] = `
<div
  class="npApp_date-with-icon"
>
  <span />
  2016-01-01
</div>
`;
